src/pages/bestProducts/BestProducts.jsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+import BestProductCardList from "../../components/bestProducts/BestProductCardList";
+
+import Title from "../../core/titles/Title";
+import fetchProduct from "../../lib/api/product";
+
+import "./best-products.css";
+import useResize from "../../lib/hooks/useResize";
+
+const BestProducts = () => {
+  const [bestProducts, setBestProducts] = useState([]);
+  const [errorMessage, setErrorMessage] = useState(null);
+  const getBestPageSize = (width) => {
+    if (1200 <= width) {
+      return 4;
+    }
+    if (768 <= width) {
+      return 2;
+    }
+    return 1;
+  };
+  const size = useResize({ getPageSize: getBestPageSize });
+
+  const handleBestProducts = async (options) => {
+    let result;
+    try {
+      setErrorMessage(null);
+      result = await fetchProduct(options);
+    } catch (error) {
+      setErrorMessage(error);
+      return;
+    }
+    const { list } = result;
+    setBestProducts(list);
```

Comment:
api 성공일때만 state에 저장하도록 할 의도라면 try 문 안에 적어주시는게 더 나아보입니다.
catch문의 return 으로 강제로 함수를 종료하는 느낌이 드네요.

```suggestion
    try {
      setErrorMessage(null);
      result = await fetchProduct(options);
      const { list } = result;
     setBestProducts(list);

    } catch (error) {
      setErrorMessage(error);
    }
```